apps/backend/src/lib/email-rendering.tsx

```diff
@@ -120,11 +120,20 @@ export async function renderEmailWithTemplate(
     "@react-email/components": "0.1.1",
     "arktype": "2.1.20",
   };
-  const output = await freestyle.executeScript(result.data, { nodeModules });
-  if (output.status === "error") {
-    return Result.error(`${output.error}`);
+  const executeResult = await freestyle.executeScript(result.data, { nodeModules });
+  if (executeResult.status === "error") {
+    return Result.error(`${executeResult.error}`);
   }
-  return Result.ok(output.data.result as { html: string, text: string, subject: string, notificationCategory: string });
+  if (!executeResult.data.result) {
+    captureError("freestyle-no-result", new StackAssertionError("No result from Freestyle", {
+      executeResult,
+      templateOrDraftComponent,
+      themeComponent,
+      options,
+    }));
+    return Result.error("An unknown error occurred while rendering this email. Please try again");
```

Comment:
this sounds like it should be a StackAssertionError instead? can we just throw the error above?